CMakeLists.txt

```diff
@@ -35,6 +35,7 @@ CPMAddPackage(
         GITHUB_REPOSITORY USCiLab/cereal
         DOWNLOAD_ONLY TRUE
 )
+add_definitions(${GCC_COVERAGE_COMPILE_FLAGS})
```

Comment:
Posso ter me confundido, mas não achei onde esta variável está sendo definida. Então não tenho certeza que está colocando essas flags só no debug. É *muito* importante que essas flags definindo o teste de cobertura não entrem no release. Testes de cobertura são importantes mas são apenas no debug. É inviável lançar um release com teste de cobertura. 

De qualquer forma, existem vários exemplos de função do cmake [aqui]( https://github.com/alandefreitas/moderncpp/tree/master/cmake). Mais especificamente, no arquivo [project_flags.cmake](https://github.com/alandefreitas/moderncpp/blob/master/cmake/functions/project_flags.cmake) tem uma macro `set_debug_booleans` que pode usar para definir as variáveis `DEBUG_MODE` e outras do tipo. Com estas, você pode fazer algo como:

```cmake
if (DEBUG_MODE)
    add_definitions(${GCC_COVERAGE_COMPILE_FLAGS})
endif()
```

Além desse problema e o problema de não saber onde está definido, temos mais um problema que é o `add_definitions` em si. Primeiro, se é flag, então não é definition. Pode ser `add_options`, setar `CMAKE_CXX_FLAGS` ou algo assim. Segundo, pro se tratar de flag, cada compilador define as suas e este código só funcionaria para o GCC. Terceiro, a boa prática é definir as flags apenas para os targets que precisam dela com `target_compile_definition` ou `target_compile_options` em vez de `add_definitions` ou `add_options`, que forçariam qualquer target, mesmo fora do seu projeto a usar estas definições.